segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/metadata/BaseMetadataStore.java

```diff
@@ -775,6 +776,23 @@ long getBufferCount() {
      */
     abstract protected CompletableFuture<Void> writeAll(Collection<TransactionData> dataList);
 
+
+    /**
+     * Retrieve all entries.
```

Comment:
You mean all entries from the `storage_metadata` table segment? If so, please provide a bit more of detail in the description of both methods.

Reply:
TableBasedMetadataStore is just one implementation of `ChunkMetadataStore` interface, there could be other implementations of metadata store backed by say ZK, data base or anything that can store KV pairs and has version based optimistic concurrency . 
The `storage_metadata` is specific to implementation  that uses internal tables.

But i get your point. I'll update the description to add more info.

Reply:
fixed.